Jenkinsfile.d/components/remoting

```diff
@@ -0,0 +1,118 @@
+        label 'release-component-remoting'
+        yamlFile 'PodTemplates.d/release-linux.yaml'
+        inheritFrom 'jnlp-linux'
+      }
+    }
+
+    options {
+      buildDiscarder logRotator(
+        artifactNumToKeepStr: '5',
+        numToKeepStr: '10'
+      )
+      disableConcurrentBuilds()
+    }
+
+    environment {
+      AZURE_VAULT_NAME              = 'prodreleasecore'
+      AZURE_VAULT_CERT              = 'prodreleasecore'
+      AZURE_VAULT_CLIENT_ID         = credentials('azure-vault-client-id')
+      AZURE_VAULT_CLIENT_SECRET     = credentials('azure-vault-client-secret')
+      AZURE_VAULT_TENANT_ID         = credentials('azure-vault-tenant-id')
+      GPG_PASSPHRASE                = credentials('release-gpg-passphrase')
+      GPG_FILE                      = 'jenkins-release.gpg'
+      MAVEN_REPOSITORY_USERNAME     = credentials('maven-repository-username')
+      MAVEN_REPOSITORY_PASSWORD     = credentials('maven-repository-password')
+      SIGN_STOREPASS                = credentials('signing-cert-pass')
+      RELEASE_PROFILE               = "components/remoting"
+    }
+
+    stages {
+      stage('Clone Remoting Git Repository') {
+        steps {
+          container('jnlp') {
+            sshagent(['release-key']) {
+              sh 'utils/release.sh --cloneReleaseGitRepository'
+            }
+          }
+        }
+      }
+      stage('Clean Release') {
+        steps {
+          container('maven') {
+            sh 'utils/release.sh --cleanRelease'
+          }
+        }
+      }
+      stage('Get Code Signing Certificate') {
+        steps {
+          container('azure-cli') {
+            sh '''
+              utils/release.sh --downloadAzureKeyvaultSecret
+              utils/release.sh --configureKeystore
+            '''
+          }
+        }
+      }
+      stage('Get GPG key') {
+        steps {
+          container('azure-cli') {
+            sh '''
+              utils/release.sh --getGPGKeyFromAzure
+            '''
+          }
+        }
+      }
+      stage('Prepare Release') {
+        steps {
+          container('maven') {
+            // Maven Release requires gpg key with password password and a certificate key with password
+            sh '''
+              utils/release.sh --configureGPG
+              utils/release.sh --configureGit
+              utils/release.sh --prepareRelease
+            '''
+
+            script {
+              def properties = readProperties file: 'release/release.properties'
+              env.RELEASE_SCM_TAG = properties['scm.tag']
+            }
+          }
+        }
+      }
+      stage('Push Commits') {
+        steps {
+          container('jnlp') {
+            sshagent(['release-key']) {
+              sh 'utils/release.sh --pushCommits'
+            }
+
+          }
+        }
+      }
+      stage('Stage Release') {
+        steps {
+          container('maven') {
+            sh '''
+              utils/release.sh --stageRelease
```

Comment:
any reason this is staged and not just done in one job?

Reply:
It's just to reuse existing script, `--stageRelease` which does run release:stage, I could easily add a new parameter named `--performRelease` to run mvn release:perform then I wouldn't need https://github.com/jenkinsci/remoting/pull/381.

Reply:
Why not do that? I can't see why you wouldn't want to just perform the release?

Reply:
It won't take too much time to add this